File: mgyoutube-multitech/api-java/src/main/java/com/djpedesen/mgyoutube/api_java/modules/UserModule.java
package com.djpedesen.mgyoutube.api_java.modules;

import java.util.List;

import com.djpedesen.mgyoutube.api_java.apimodel.User;
import com.djpedesen.mgyoutube.api_java.apimodel.UserCredential;

public interface UserModule {

	User authUser(UserCredential userCredential);

	User createUpdateUser(User user);

	User createUser(User user);

	User updateUser(String userId, User user);

	User getUser(String username);

	User removeUser(String username);

	User addUpdateChildToParent(String parentUsername, User childUser) throws UserNotFoundException;

	List<User> getChildrenForParent(String parentUsername) throws UserNotFoundException;

}

File: mgyoutube-multitech/api-go/src/modules/usermodule.go
package modules

import "apimodel"

// UserModule - defines the interface for a UserModule
type UserModule interface {
	AuthUser(userCredential *apimodel.UserCredential) *apimodel.User

	GetUserByUsername(username string) *apimodel.User

	CreateUpdateUser(user *apimodel.User) *apimodel.User
	CreateUser(user *apimodel.User) *apimodel.User
	UpdateUser(userID string, user *apimodel.User) *apimodel.User
	RemoveUser(username string) *apimodel.User

	GetChildrenForParent(parentUsername string) ([]*apimodel.User, error)
	AddUpdateChildToParent(parentUsername string, childUser *apimodel.User) (*apimodel.User, error)
}
